File: 백준/Silver/1920. 수 찾기/수 찾기.java
import java.io.*;
import java.util.*;

public class Main {
    static int N;
    static int[] arr;
    public static void main(String[] args) throws IOException {
        BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
        StringBuilder sb = new StringBuilder();
        N = Integer.parseInt(br.readLine());
        arr = new int[N];
        StringTokenizer st = new StringTokenizer(br.readLine());
        for(int i = 0; i < N; i++) {
            arr[i] = Integer.parseInt(st.nextToken());
        }
        Arrays.sort(arr);
        int M = Integer.parseInt(br.readLine());
        st = new StringTokenizer(br.readLine());
        for(int i = 0; i < M; i++) {
            sb.append(binarySearch(Integer.parseInt(st.nextToken())) ? "1\n" : "0\n");
        }
        System.out.println(sb.toString().trim());
    }

    static boolean binarySearch(int target) {
        int left = 0;
        int right = N - 1;
        while(left <= right) {
            int mid = (left + right) / 2;
            if(arr[mid] == target) return true;
            else if(arr[mid] < target) left = mid + 1;
            else right = mid - 1;
        }
        return false;
    }
}

File: 백준/Silver/1920. 수 찾기/수 찾기.js
let [N, arr, M, input] = require('fs').readFileSync('/dev/stdin').toString().trim().split('\n').map(line => line.trim());
N = Number(N);
M = Number(M);
arr = arr.split(' ').map(Number).sort((a, b) => a - b);
input = input.split(' ').map(Number);
const ans = [];
for(let i = 0; i < input.length; i++) {
  ans.push(binarySearch(input[i]) ? 1 : 0);
}
console.log(ans.join('\n'));

function binarySearch(target) {
  let left = 0;
  let right = N - 1;
  while(left <= right) {
    let mid = Math.floor((left + right) / 2);
    if(arr[mid] === target) return true;
    else if(arr[mid] > target) right = mid - 1;
    else left = mid + 1;
  }
  return false;
}